### DISCUSSION TITLE
Is there a API spec for the ZPM registry

### DISCUSSION THREAD

**Discussion by johnmcbride** [intersystems/ipm#409] [Q&A]
I'm interested in building a private registry to host some private packages with integration into artifact servers (like artifactory). Is there a API spec for the current ZPM registry?

**Comment by evshvarov** [comment]
hi @johnmcbride ! You can use [zpm-registry](https://openexchange.intersystems.com/package/zpm-registry) for that.
just deploy vanilla iris with installed zpm-registry package. It deploys REST-API that will serve IPM registry for you.

**Reply by evshvarov** [reply]
[Related article](https://community.intersystems.com/post/setting-your-own-intersystems-objectscript-package-manager-registry)
